homeassistant/components/skybell/light.py

```diff
@@ -1,90 +1,76 @@
 """Light/LED support for the Skybell HD Doorbell."""
 from __future__ import annotations
 
+from typing import Any
+
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS,
     ATTR_HS_COLOR,
-    SUPPORT_BRIGHTNESS,
-    SUPPORT_COLOR,
+    COLOR_MODE_BRIGHTNESS,
+    COLOR_MODE_HS,
     LightEntity,
 )
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 import homeassistant.util.color as color_util
 
-from . import DOMAIN as SKYBELL_DOMAIN, SkybellDevice
+from . import SkybellEntity
+from .const import DOMAIN
+from .coordinator import SkybellDataUpdateCoordinator
 
 
-def setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
-    """Set up the platform for a Skybell device."""
-    skybell = hass.data[SKYBELL_DOMAIN]
-
-    sensors = []
-    for device in skybell.get_devices():
-        sensors.append(SkybellLight(device))
-
-    add_entities(sensors, True)
-
-
-def _to_skybell_level(level):
-    """Convert the given Home Assistant light level (0-255) to Skybell (0-100)."""
-    return int((level * 100) / 255)
+    """Set up Skybell switch."""
+    async_add_entities(
+        SkybellLight(coordinator)
+        for coordinator in hass.data[DOMAIN][entry.entry_id].values()
+    )
 
 
-def _to_hass_level(level):
-    """Convert the given Skybell (0-100) light level to Home Assistant (0-255)."""
-    return int((level * 255) / 100)
+class SkybellLight(SkybellEntity, LightEntity):
+    """A light implementation for Skybell devices."""
 
+    _attr_supported_color_modes = {COLOR_MODE_BRIGHTNESS, COLOR_MODE_HS}
+    coordinator: SkybellDataUpdateCoordinator
 
-class SkybellLight(SkybellDevice, LightEntity):
-    """A binary sensor implementation for Skybell devices."""
-
-    def __init__(self, device):
+    def __init__(
+        self,
+        coordinator: SkybellDataUpdateCoordinator,
+    ) -> None:
         """Initialize a light for a Skybell device."""
-        super().__init__(device)
-        self._name = self._device.name
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
+        super().__init__(coordinator)
+        self._attr_name = coordinator.name
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{self.name}"
 
-    def turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn on the light."""
         if ATTR_HS_COLOR in kwargs:
```

Comment:
It looks like this won't work correctly if ATTR_BRIGHTNESS and ATTR_HS_COLOR are both passed at the same time